Remotion/Validation/Core/Implementation/GenericTypeAwareValidatedTypeResolverDecorator.cs

```diff
@@ -46,9 +49,21 @@ public IValidatedTypeResolver InnerResolver
     {
       ArgumentUtility.CheckNotNull ("collectorType", collectorType);
 
-      if (TypeExtensions.CanAscribeTo (collectorType, typeof(IValidationRuleCollector<>)))
-        return TypeExtensions.GetAscribedGenericArguments (collectorType, typeof(IValidationRuleCollector<>)).Single();
-      return _validatedTypeResolver.GetValidatedType (collectorType);
+      return s_canAscribeToGenericArgumentsOfIValidationRuleCollectorCache.GetOrAdd (
+              collectorType,
+              static (t, resolver) =>
+              {
+                var canAscribeTo = t.CanAscribeTo (typeof (IValidationRuleCollector<>));
```

Comment:
We can do a test for IValidationRuleCollector first.

Reply:
What do you mean with test for `IValidationRuleCollector`?
If i check it the cache would be useless, because it would invoke `CanAscribeTo` all the time🤔

Reply:
`typeof (IValidationRuleCollector).IsAssignableFrom(t) &&  t.CanAscribeTo (typeof (IValidationRuleCollector<>))`